File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
               int answer = 0;

        for (int i = 0; i < discount.length - 9; i++) {
            int j;
            for (j = 0; j < want.length; j++) {
                int check = 0;
                for (int k = i; k < i + 10; k++) {
                    if (discount[k].equals(want[j])) check++;
                }
                if (check < number[j]) {
                    break;
                }
            }
            if (j == want.length) answer++;
        }

        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    
    for i in range(len(discount)-9):
        ok = True
        for j in range(len(want)):
            if discount[i:i+10].count(want[j]) != number[j]:
                ok = False
                break
        if ok:
            answer += 1
                
    return answer